docs/cudf/source/user_guide/groupby.md

```diff
@@ -0,0 +1,268 @@
+7. `GroupBy.groups` API that returns a mapping of group keys to row
+   labels
+8. `GroupBy.apply` API for performing arbitrary operations on each
+   group. Note that this has very limited functionality compared to the
+   equivalent Pandas function. See the section on
+   [apply](#groupby-apply) for more details.
+9. `GroupBy.pipe` similar to
+   [Pandas](https://pandas.pydata.org/pandas-docs/stable/user_guide/groupby.html#piping-function-calls).
+
+## Grouping
+
+A GroupBy object is created by grouping the values of a `Series` or
+`DataFrame` by one or more columns:
+
+```python
+import cudf
+
+>>> df = cudf.DataFrame({'a': [1, 1, 1, 2, 2], 'b': [1, 1, 2, 2, 3], 'c': [1, 2, 3, 4, 5]})
+>>> df
+>>> gb1 = df.groupby('a')  # grouping by a single column
+>>> gb2 = df.groupby(['a', 'b'])  # grouping by multiple columns
+>>> gb3 = df.groupby(cudf.Series(['a', 'a', 'b', 'b', 'b']))  # grouping by an external column
+```
+
+:::{warning}
+cuDF uses `sort=False` by default to achieve better performance, which provides no gaurentee to the group order in outputs. This deviates from Pandas default behavior.
+
+For example:
+
+```python
+>>> df = cudf.DataFrame({'a' : [2, 2, 1], 'b' : [42, 21, 11]})
+>>> df.groupby('a').sum()
+   b
+a
+2  63
+1  11
+>>> df.to_pandas().groupby('a').sum()
+   b
+a
+1  11
+2  63
+```
+
+Setting `sort=True` will produce Pandas-like output, but with some performance penalty:
+
+```python
+>>> df.groupby('a', sort=True).sum()
+   b
+a
+1  11
+2  63
+```
+:::
+
+### Grouping by index levels
+
+You can also group by one or more levels of a MultiIndex:
+
+```python
+>>> df = cudf.DataFrame(
+...     {'a': [1, 1, 1, 2, 2], 'b': [1, 1, 2, 2, 3], 'c': [1, 2, 3, 4, 5]}
+... ).set_index(['a', 'b'])
+...
+>>> df.groupby(level='a')
+```
+
+### The `Grouper` object
+
+A `Grouper` can be used to disambiguate between columns and levels
+when they have the same name:
+
+```python
+>>> df
+   b  c
+b
+1  1  1
+1  1  2
+1  2  3
+2  2  4
+2  3  5
+>>> df.groupby('b', level='b')  # ValueError: Cannot specify both by and level
+>>> df.groupby([cudf.Grouper(key='b'), cudf.Grouper(level='b')])  # OK
+```
+
+## Aggregation
+
+Aggregations on groups is supported via the `agg` method:
+
+```python
+>>> df
+   a  b  c
+0  1  1  1
+1  1  1  2
+2  1  2  3
+3  2  2  4
+4  2  3  5
+>>> df.groupby('a').agg('sum')
+   b  c
+a
+1  4  6
+2  5  9
+>>> df.groupby('a').agg({'b': ['sum', 'min'], 'c': 'mean'})
+    b        c
+  sum min mean
+a
+1   4   1  2.0
+2   5   2  4.5
+>>> df.groupby("a").corr(method="pearson")
+          b          c
+a
+1 b  1.000000  0.866025
+  c  0.866025  1.000000
+2 b  1.000000  1.000000
+  c  1.000000  1.000000
+```
+
+The following table summarizes the available aggregations and the types
+that support them:
+
+```{eval-rst}
+.. table::
+    :class: special-table
+
+    +------------------------------------+-----------+------------+----------+---------------+--------+----------+------------+-----------+
```

Comment:
This table gets rendered incorrectly and I'm investigating:

<img width="1272" alt="Screen Shot 2022-04-26 at 7 38 09 AM" src="https://user-images.githubusercontent.com/3190405/165292014-8a0eece3-5091-48e4-b52f-742c9f3e6da6.png">